apis/placement/v1alpha1/stagedupdate_types.go

```diff
@@ -346,6 +368,12 @@ const (
 	// - "False" means the staged update run is not waiting between stages.
 	// - "Unknown" means it is unknown.
 	StagedUpdateRunConditionWaitingBetweenStage StagedUpdateRunConditionType = "WaitingBetweenStage"
```

Comment:
do we need the "Running" and "WaitingBetweenStage" conditions? they could be "reason" when the completed is "false".

"Running", "WaitingBetweenStage" and "Completed" all these conditions are delivering some duplicate information.

Reply:
With "running" and "waittingBetweenStage" conditions, we will get the information like last time the stage has completed. If we collapse them into one condition then we lost a bit of the history.

Reply:
Do we really need both conditions and state? State serves the same purpose as conditions. State contains starttime and endtime.

RunStatus {
  runState State
  stages []StageStatus
  deletionStage StageStatus
}

StageStatus {
  stageName string
  stageState State
  clusters []ClusterStatus
  afterStageTasks []TaskStatus
}

ClusterStatus {
  clusterName string
  clusterState State
}

TastkStatus {
  taskType 
  taskState State
}

Reply:
It's [k8s convention](https://github.com/zecke/Kubernetes/blob/master/docs/devel/api-conventions.md#typical-status-properties) in the status to use condition to show the state transition which is pretty much equivalent to the runState. I think using state is more efficient and acceptable as an internal field but we better to keep condition as the first layer fields status. If we really want to unify them then they all should be conditions like in CPR.

Reply:
agree with Liqian about the detailed stage information.  They are already part of the StageStatus. 

 For the conditions in the "StagedUpdateRunStatus", can we treat it as overall condition/status? so that user can query this condition to know about whether the updateRun has completed or not.